dotnet/src/MicroFocus.CafApi.QueryBuilder.Vqll.Parser.SystemJson/SystemJsonFilterReader.cs

```diff
@@ -0,0 +1,537 @@
+        }
+
+        private Filter<string> CreateContainsFilter(JsonArray node)
+        {
+            return FilterFactory.Contains(node[1].GetValue<string>(), node[2].GetValue<string>());
+        }
+
+        private Filter<string> CreateStartsWithFilter(JsonArray node)
+        {
+            return FilterFactory.StartsWith(node[1].GetValue<string>(), node[2].GetValue<string>());
+        }
+
+        private Filter<string> CreateLikeFilter(JsonArray node)
+        {
+            return FilterFactory.Like(node[1].GetValue<string>(), GetLikeTokens(GetFilterArgumentsElements(node)));
+        }
+
+        private IEnumerator<JsonNode> GetFilterArgumentsElements(JsonArray node)
+        {
+            IEnumerator<JsonNode> iter = node.GetEnumerator();
+            iter.MoveNext(); // filter type
+            iter.MoveNext(); // field name
+            return iter;
+        }
+
+        private Filter<string> CreateBetweenFilter(JsonArray node)
+        {
+            bool isStartANumber = false;
+            Object start;
+            if (node[2] == null)
+            {
+                start = null;
+            }
+            else
+            {
+                switch (node[2].GetValue<JsonElement>().ValueKind)
+                {
+                    case JsonValueKind.Number:
+                        start = node[2].GetValue<long?>();
+                        isStartANumber = true;
+                        break;
+                    case JsonValueKind.String:
+                        start = node[2].GetValue<string>();
+                        break;
+                    default:
+                        throw new ArgumentException("Unexpected start value type in 'between' filter: " + node[2].GetValue<JsonElement>().ValueKind);
+                }
+            }
+            bool isEndANumber = false;
+            Object end;
+            if (node[3] == null)
+            {
+                end = null;
+            }
+            else
+            {
+                switch (node[3].GetValue<JsonElement>().ValueKind)
+                {
+                    case JsonValueKind.Number:
+                        end = node[3].GetValue<long?>();
+                        isEndANumber = true;
+                        break;
+                    case JsonValueKind.String:
+                        end = node[3].GetValue<string>();
+                        break;
+                    default:
+                        throw new ArgumentException("Unexpected end value type in 'between' filter: " + node[3].GetValue<JsonElement>().ValueKind);
+                }
+            }
+
+            if (isStartANumber || isEndANumber)
+            {
+                Filter<string> betweenFilter =
+                    start == null
+                    ? FilterFactory.Between(node[1].GetValue<string>(), null, Convert.ToInt64(end))
+                    : FilterFactory.Between(node[1].GetValue<string>(), Convert.ToInt64(start), null);
```

Comment:
I would have thought that you should just cast here.

```suggestion
                    ? FilterFactory.Between(node[1].GetValue<string>(), null, (long)end)
                    : FilterFactory.Between(node[1].GetValue<string>(), (long)start, null);
```

`Convert.ToInt64()` would have lots of unnecessary functionality.